rclcpp/include/rclcpp/subscription_options.hpp

```diff
@@ -0,0 +1,167 @@
+// Copyright 2019 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef RCLCPP__SUBSCRIPTION_OPTIONS_HPP_
+#define RCLCPP__SUBSCRIPTION_OPTIONS_HPP_
+
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "rclcpp/visibility_control.hpp"
+
+namespace rclcpp
+{
+
+struct SubscriptionEventCallbacks
+{
+  // NOTE(emersonknapp): empty placeholder for future additions of QoS Event callbacks
+};
+
+template<typename Alloc = std::allocator<void>>
+class SubscriptionOptions
+{
+public:
+  /// Create SubscriptionOptions with default values, optionally specifying the allocator to use.
+  /**
+   * Default values for the node options:
```

Comment:
Let's be careful with listing default values like this in docstrings, this rot easily.